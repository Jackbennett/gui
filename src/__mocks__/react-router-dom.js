import React from 'react';
import { compose, setDisplayName } from 'recompose';
import * as rrd from 'react-router-dom';
// Just render plain div with its children
// eslint-disable-next-line no-import-assign
rrd.BrowserRouter = compose(setDisplayName('BrowserRouter'))(({ children }) => <div>{children}</div>);

module.exports = rrd;
