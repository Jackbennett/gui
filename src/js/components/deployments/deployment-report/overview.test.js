import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import moment from 'moment';
import momentDurationFormatSetup from 'moment-duration-format';
import DeploymentOverview from './overview';

let dateMock;
momentDurationFormatSetup(moment);

describe('DeploymentOverview Component', () => {
  beforeEach(() => {
    const mockDate = new Date('2019-01-01T13:00:00.000Z');
    const _Date = Date;
    global.Date = jest.fn(() => mockDate);
    global.Date.parse = _Date.parse;
    global.Date.now = _Date.now;
    dateMock = jest.spyOn(global, 'Date').mockImplementation(() => mockDate);
  });

  afterEach(() => {
    dateMock.mockRestore();
  });

  it('renders correctly', () => {
    const deployment = {
      name: 'test deployment',
      artifact_name: 'test',
      created: '2019-01-01',
      devices: {},
      finished: '2019-01-01',
      stats: {}
    };
    const creationMoment = moment();
    const elapsedMoment = moment();
    const duration = moment.duration(elapsedMoment.diff(creationMoment));
    const tree = renderer
      .create(
        <MemoryRouter>
          <DeploymentOverview
            allDevices={[]}
            deployment={deployment}
            deviceCount={0}
            duration={duration}
            onAbortClick={jest.fn}
            onRetryClick={jest.fn}
            viewLog={jest.fn}
          />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
