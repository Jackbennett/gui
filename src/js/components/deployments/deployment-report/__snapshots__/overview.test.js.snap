// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeploymentOverview Component renders correctly 1`] = `
<div>
  <div
    className="report-container"
  >
    <div
      className="deploymentInfo two-columns"
    >
      <div>
        <div
          onClick={[Function]}
          style={
            Object {
              "marginBottom": -15,
            }
          }
        >
          <li
            className="MuiListItem-root attributes MuiListItem-gutters Mui-disabled"
            disabled={true}
          >
            <div
              className="MuiListItemText-root MuiListItemText-multiline"
            >
              <span
                className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
              >
                Release:
              </span>
              <div
                className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary"
                title={
                  <Link
                    style={
                      Object {
                        "fontWeight": "500",
                      }
                    }
                    to="/releases/test"
                  >
                    test
                  </Link>
                }
              >
                <div>
                  <span
                    className=" "
                    style={Object {}}
                  >
                    <a
                      href="/releases/test"
                      onClick={[Function]}
                      style={
                        Object {
                          "fontWeight": "500",
                        }
                      }
                    >
                      test
                    </a>
                  </span>
                   
                </div>
              </div>
            </div>
          </li>
        </div>
        <div
          onClick={[Function]}
          style={
            Object {
              "marginBottom": -15,
            }
          }
        >
          <li
            className="MuiListItem-root attributes MuiListItem-gutters Mui-disabled"
            disabled={true}
          >
            <div
              className="MuiListItemText-root MuiListItemText-multiline"
            >
              <span
                className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
              >
                Device group:
              </span>
              <div
                className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary"
                title="test deployment"
              >
                <div>
                  <span
                    className=" "
                    style={Object {}}
                  >
                    test deployment
                  </span>
                   
                </div>
              </div>
            </div>
          </li>
        </div>
      </div>
      <div
        onClick={[Function]}
        style={
          Object {
            "marginBottom": -15,
          }
        }
      >
        <li
          className="MuiListItem-root attributes MuiListItem-gutters Mui-disabled"
          disabled={true}
        >
          <div
            className="MuiListItemText-root MuiListItemText-multiline"
          >
            <span
              className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
            >
              Status:
            </span>
            <div
              className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary"
              title={
                <div>
                  Finished 
                </div>
              }
            >
              <div>
                <span
                  className=" "
                  style={Object {}}
                >
                  <div>
                    Finished 
                  </div>
                </span>
                 
              </div>
            </div>
          </div>
        </li>
      </div>
    </div>
  </div>
</div>
`;
